Sources/SPMBuildCore/PluginInvocation.swift

```diff
@@ -560,6 +542,31 @@ public extension PluginTarget {
             }
         })
     }
+
+    func processAccessibleTools(packageGraph: PackageGraph, fileSystem: FileSystem, environment: BuildEnvironment, for hostTriple: Triple, builtToolHandler: (_ name: String, _ path: RelativePath) throws -> AbsolutePath?) throws -> (toolNamesToPaths: [String: AbsolutePath], toolNamesToTriples: [String: [String]]) {
```

Comment:
Would this be easier to reason about if it returned `[String: (path: AbsolutePath, triples: [String])]`?

Reply:
i.e. instead of parallel dictionaries, just a mapping from a toolname to the information about that tool name?

Reply:
I'm more and more leaning towards that, but I think I'll do that separately from this PR. Just because I think it would be better to make that change everywhere, not separate into the two dictionaries at the callsite and at that point it starts feeling like a separate change.

Reply:
https://github.com/apple/swift-package-manager/pull/5976